src/analysis/retail/mage/arcane/core/ArcaneBarrage.tsx

```diff
@@ -33,6 +34,7 @@ export default class ArcaneBarrage extends Analyzer {
   hasArcaneSoul: boolean = this.selectedCombatant.hasTalent(TALENTS.MEMORY_OF_ALAR_TALENT);
 
   barrageCasts: ArcaneBarrageCast[] = [];
+  allCasts: CastEvent[] = [];
```

Comment:
It looks like you are creating a separate object to house every cast so you can figure out if Touch of the Magi was cast directly after Barrage. You can get this using CastLinkNormalizer. If you link Barrage Casts to Touch of the Magi Casts within a short period of time then you can use GetRelatedEvents to check to see if a match was found and then can link them to the barrageCasts object that already exists.